### PR TITLE
Add Base CLI with Velero Init

### PR DESCRIPTION
**Pull request by shaneutt**
The purpose of this PR is to add Velero initialization to provide a fallback S3 backend for Velero if no other is provided by the end user.

### CODE REVIEW

File: cmd/velero.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		fallback, err := utils.IsVeleroBackendSetToFallback()
+		if err != nil {
+			return err
+		}
+
+		if fallback {
+			log.Println("WARNING: Velero is configured for fallback backend")
+			log.Println("WARNING: The fallback configuration should not be used in production")
+
+			m, err := utils.NewMinioBuilder("velero")
```
Comment (shaneutt) [inline cmd/velero.go]:
```suggestion
			m, err := utils.NewMinioBuilder(utils.DefaultVeleroNamespace)
```

File: pkg/utils/doc.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/*
+Package utils provides general utility functionality
+
+
+Overview
+
+These utilities include tooling for interacting with Kubernetes as well as deploying Minio cluster.
```
Comment (shaneutt) [inline pkg/utils/doc.go]:
```suggestion
These utilities include tooling for interacting with Kubernetes as well as initializing Velero
```

File: pkg/utils/minio.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+func (m *MinioBuilder) deployMinioInstance() error {
+	i := &miniov1beta1.MinIOInstance{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "minio",
+		},
+		Spec: miniov1beta1.MinIOInstanceSpec{
+			Replicas: 4,
+			Version:  "RELEASE.2019-06-19T18-24-42Z",
```
Comment (joejulian) [inline pkg/utils/minio.go]: Make configurable by ENV

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+WORKDIR /go/src/github.com/kubeaddons/init-containers
+
+ENV GO111MODULE on
+
+# ------------------------------------------------------------------------------
+# Build Image - Build
+# ------------------------------------------------------------------------------
+
+COPY go.mod /go/src/github.com/kubeaddons/init-containers/
```
[the remark this answers is not in the document]
Comment (shaneutt) [reply Dockerfile]: I really like to be explicit about what I'm copying in, as I like to know that I'm only copying what I need (for instance, there's no need for me to copy in the `test/` directory). It saves a tiny lit bit of disk space, and also can be used as documentation to see what directories are actually required to build. Can I persuade you on this?

File: pkg/utils/minio.go [hunk 5]
```diff
@@ -0,0 +1,155 @@
+package utils
```
[the remark this answers is not in the document]
Comment (shaneutt) [reply pkg/utils/minio.go]: :+1:

File: pkg/utils/velero.go [hunk 6]
```diff
@@ -0,0 +1,41 @@
+package utils
```
[the remark this answers is not in the document]
Comment (shaneutt) [reply pkg/utils/velero.go]: :+1:

File: Dockerfile (hunk 4 above)
[the remark this answers is not in the document]
Comment (joejulian) [reply Dockerfile]: It's discarded so no disk space is saved. It requires extra human maintenance as the code changes which leaves open the opportunity for error. This isn't a hard blocker, but it's just unnecessary complexity that I'd prefer to have simplified.

[the remark this answers is not in the document]
Comment (shaneutt) [reply Dockerfile]: @joejulian the build image layers are cached too, it _will_ impact disk space albeit totally negligibly.

These little details are unimportant though, so I've gone ahead and changed it :+1:

File: cmd/velero.go [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+	RunE: func(cmd *cobra.Command, args []string) error {
+		fallback, err := utils.IsVeleroBackendSetToFallback()
+		if err != nil {
+			return err
+		}
+
+		if fallback {
+			log.Println("WARNING: Velero is configured for fallback backend")
+			log.Println("WARNING: The fallback configuration should not be used in production")
+
```
[the remark this answers is not in the document]
Comment (shaneutt) [reply cmd/velero.go]: :+1: sounds good

File: pkg/utils/minio.go (hunk 3 above)
Comment (shaneutt) [reply pkg/utils/minio.go]: :+1: sounds good

**Comment by shaneutt** [comment]
Just finishing up tests, password management, and making sure everything is working in Travis and then this will be ready to go.

File: .travis.yml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    - $HOME/gopath/pkg/mod
+    - /gopath/pkg/mod
+
+jobs:
+  include:
+    - stage: Unit Tests
+      script: make test
+    - stage: Integration Tests
+      before_script:
+        - curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
```
Comment (shaneutt) [inline .travis.yml]: This line isn't needed anymore, it was from a time when I was using `kubectl` for prototyping.
```suggestion
```

File: test/utils/kubernetes.go [hunk 9]
```diff
@@ -0,0 +1,26 @@
+package utils
```
Comment (shaneutt) [inline test/utils/kubernetes.go]: This package isn't needed anymore, I'm going to remove it.

**Pull request merged**